lib/logthrsource/tests/test_logthrfetcherdrv.c

```diff
@@ -243,3 +243,114 @@ Test(logthrfetcherdrv, test_reconnect)
 
   destroy_test_threaded_fetcher(s);
 }
+
+static LogThreadedFetchResult
+_fetch_for_try_again_test(LogThreadedFetcherDriver *s)
+{
+  TestThreadedFetcherDriver *self = (TestThreadedFetcherDriver *) s;
+
+  if (__once())
+    {
+      return (LogThreadedFetchResult)
+      {
+        THREADED_FETCH_TRY_AGAIN, NULL
+      };
+    }
+
+  g_mutex_lock(self->lock);
+  if (self->num_of_messages_to_generate <= 0)
+    {
+      g_cond_signal(self->cond);
+      g_mutex_unlock(self->lock);
+      return (LogThreadedFetchResult)
+      {
+        THREADED_FETCH_ERROR, NULL
+      };
+    }
+
+  LogMessage *msg = create_sample_message();
+
+  self->num_of_messages_to_generate--;
+  g_mutex_unlock(self->lock);
+
+  return (LogThreadedFetchResult)
+  {
+    .result = THREADED_FETCH_SUCCESS,
+    .msg = msg
+  };
+}
+
+Test(logthrfetcherdrv, test_try_again)
+{
+  TestThreadedFetcherDriver *s = create_threaded_fetcher();
+
+  s->num_of_messages_to_generate = 1;
+  s->super.time_reopen = 10;
+  s->super.fetch = _fetch_for_try_again_test;
+
+  time_t start = time(NULL);
+
+  start_test_threaded_fetcher(s);
+  wait_for_messages(s);
+  stop_test_threaded_fetcher(s);
+
+  // Should not pass time_reopen in case of try_again
+  cr_assert(time(NULL) - start < 2);
```

Comment:
Is `time()` reliable enough for a test case? Shouldn't we use monotonic time?

Reply:
I think it is reliable enough for a unit test, though it does not introduce much complexity to use monothonic time (I am a little concerned about compatibility when porting to non-linux platforms though). In the end, I changed to monothonic time.